### DISCUSSION TITLE
Stripping leading whitespace in multiline templates

### DISCUSSION THREAD

**Discussion by unikitty37** [twpayne/chezmoi#3866] [Q&A] [answered]
I have a template like this:

```
{{ if eq .chezmoi.os "darwin" -}}
  export A=1
  export B=2
  export C-3
{{ end }}
```

but this seems to treat it as a single string `  export A=1\n  export B=2\n  export C=3\n`, meaning it just strips the very first two spaces and results in this substituted text:

```
export A=1
  export B=2
  export C=3
```

Is there a way to get it to strip whitespace on _each_ line? I really don't want to have to put an if…end block for each line!

**Comment by twpayne** [comment, marked as answer]
> Is there a way to get it to strip whitespace on _each_ line? I really don't want to have to put an if…end block for each line!

You need to remove the whitespace yourself, i.e.

```
{{ if eq .chezmoi.os "darwin" -}}
export A=1
export B=2
export C-3
{{ end }}
```

See the [`text/template` docs on whitespace](https://pkg.go.dev/text/template#hdr-Text_and_spaces) for more information.

**Reply by unikitty37** [reply]
Thanks — I was hoping there'd be a way to indent the block and have it look neat in the template file, like Ruby's `<<~` operator. Guss not 🙂
